src/parents.js

```diff
@@ -9,14 +10,19 @@ const parentMap = new WeakMap();
  * @param {*} node
  * @param {object} [options]
  * @param {boolean} [options.force] Overwrite existing `parent` properties
+ * @throws {Error} If one of the nodes is not found in its parent
  */
 export function setAll (node, options ) {
 	walk(node, (node, property, parent) => {
-		let ret = set(node, parent, options);
-
-		if (ret === false) {
-			// We assume that if the node already has a parent, its subtree will also have parents
-			return false;
+		try {
+			const ret = set(node, parent, options);
+			if (ret === false) {
+				// We assume that if the node already has a parent, its subtree will also have parents
+				return false;
+			}
+		}
+		catch (e) {
+			throw new Error(`Could not set parent for node of type ${node.type} at ${property} in parent of type ${parent.type}`);
```

Comment:
I’m open to be convinced otherwise but not sure why we need to `try...catch` here? We are not providing much of a higher abstraction than the error in `set()`, it seems like we're just saying the same thing, if anything with less useful detail.
I don't think something like "Could not set parent for node of type MemberExpression at callee in parent of type CallExpression" would help debugging much. At least if we simply let the error from `set()` pass through, we could just pass the context in [`cause`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error#cause).

Reply:
Sure! I was also thinking about the same thing when implementing, but added it so there wouldn't be any uncaught errors. The only time it would end up throwing an uncaught error is when the tree is somehow malformed, since it otherwise walks it (implying the parent-child relationship must exist)